### PR TITLE
opentelemetry

### PR DESCRIPTION
**Pull request by liyongjing**
go.opentelemetry.io v0.0.0-20191101045625-a71ec85f5e10

### CODE REVIEW

File: internal/platform/trace/trace.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+		return err
+	}
+	global.SetTraceProvider(tp)
+	return nil
+}
+
+// NewSpan start a span with trace
+func NewSpan(ctx context.Context, name string) context.Context {
+	ctx, span := global.Tracer("main").Start(ctx, name)
+	defer span.End()
```
Comment (ardan-bkennedy) [inline internal/platform/trace/trace.go:44]: This ends the span immediately? Doesn't think call need to be made by the caller on the returned context?

Comment (ardan-bkennedy) [reply internal/platform/trace/trace.go:44]: I cloned your fork and ran the code. I am getting full spans. I don't understand how this could be working?

Comment (ardan-bkennedy) [reply internal/platform/trace/trace.go:44]: Cloning your repo is not giving me this code. I forgot to switch branches.

Comment (ardan-bkennedy) [reply internal/platform/trace/trace.go:44]: Ran the correct code. That call to End needs to happen by the caller. Please make that change.

Comment (ardan-bkennedy) [reply internal/platform/trace/trace.go:44]: I'm making the change on your fork and I will try to submit a PR.

Comment (liyongjing) [reply internal/platform/trace/trace.go:44]: I means don't be scattered anywhere for go.opentelemetry.io/otel/sdk/trace import, so using      NewSpan func, I modify that two return value.

Comment (ardan-bkennedy) [reply internal/platform/trace/trace.go:44]: I am about to push some code to new.  Removed the New function, it was not necessary.

Comment (ardan-bkennedy) [reply internal/platform/trace/trace.go:44]: I have a picture of the zipkin report I can show you before / after.

Comment (liyongjing) [reply internal/platform/trace/trace.go:44]: ok, you do

Comment (ardan-bkennedy) [reply internal/platform/trace/trace.go:44]: I didn't fork your fork. Can I have rights on this fork?

Comment (ardan-bkennedy) [reply internal/platform/trace/trace.go:44]: I think you were right because your trace shows individual times per span. How does that work?

Comment (ardan-bkennedy) [reply internal/platform/trace/trace.go:44]: I am very very confused. Sorry.

func NewSpan(ctx context.Context, name string) context.Context {
	ctx, span := global.Tracer("main").Start(ctx, name)
	defer span.End()
	return ctx
}
My guess is your trace is showing how long it takes to execute those two lines of code?

Comment (ardan-bkennedy) [reply internal/platform/trace/trace.go:44]: That can't be right. I have code I think is right. I can push it to your repo or maybe what I will do it merge this into service. Then you can look at it?

Comment (liyongjing) [reply internal/platform/trace/trace.go:44]: sales-api 
......687μs : internal.platform.web
 sales-api 
......2μs : internal.mid.logger
 sales-api 
......1μs : internal.mid.errors
 sales-api 
......1μs : internal.mid.metrics
 sales-api 
......13μs : handlers.check.health
sales-api 
......
I run curl -H "Authorization: Bearer ${TOKEN}" http://localhost:3000/v1/health show above, defer statement execute at function stack finished.

Comment (liyongjing) [reply internal/platform/trace/trace.go:44]: > I am very very confused. Sorry.
> 
> func NewSpan(ctx context.Context, name string) context.Context {
> ctx, span := global.Tracer("main").Start(ctx, name)
> defer span.End()
> return ctx
> }
> My guess is your trace is showing how long it takes to execute those two lines of code?

this is wrong...span scope becomes smaller

Comment (ardan-bkennedy) [reply internal/platform/trace/trace.go:44]: Ok, cool. I am going to move the changes to my repo, then you can take a look. I appreciate the work on this.

Comment (ardan-bkennedy) [reply internal/platform/trace/trace.go:44]: Take a look at this

https://github.com/ardanlabs/service/pull/123

Comment (liyongjing) [reply internal/platform/trace/trace.go:44]: you right for this pr, Probability property in Zipkin struct  has discarded

Comment (ardan-bkennedy) [reply internal/platform/trace/trace.go:44]: Check the other PR moving forward.

**Pull request closed without merging**